nucleus/deployment/admin/src/main/java/org/glassfish/deployment/admin/DeploymentCommandUtils.java

```diff
@@ -89,6 +89,31 @@
     final private static String COPY_IN_PLACE_ARCHIVE_PROP_NAME = "copy.inplace.archive";
     
     private static final List<String> LIST_CONTAINING_DOMAIN = new ArrayList<String>(Arrays.asList(DeploymentUtils.DOMAIN_TARGET_NAME));
+    
+    /** Temporary variable for calculating time passed when deploying an app. */
+    private static long startTime = System.nanoTime();
+    
+    /**
+     * Starts a timer. You should call {@link getTimePassedInSeconds()} to then 
+     * get the time that has passed since the timer started.
+     */
+    public static void startTimer() {
+        startTime = System.nanoTime();
+    }
+    
+    /**
+     * Gets the time (in seconds) that has passed since the timer was started. 
+     * If the timer hasn't been started yet you may get strange results.
+     * The time returned is to millisecond accuracy.
```

Comment:
Because I thought that System.currentTimeMillis() is more a measure of wall time than time passed. It's not always guaranteed to be positive. Having said that it might not be necessary for low accuracy timings like this. What do you think?